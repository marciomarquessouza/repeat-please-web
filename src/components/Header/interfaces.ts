export interface IHeaderProps {
	onLoginClick: () => void;
}
