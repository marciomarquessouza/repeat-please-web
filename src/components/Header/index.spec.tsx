import React from 'react';
import { shallow } from 'enzyme';
import { Header } from '.';
import { IHeaderProps } from './interfaces';
import { findByDataTest } from '../../utils/tests/findByDataTest';

const setup = (props: IHeaderProps) => shallow(<Header {...props} />);

describe('#Header', () => {
	it('renders properly', () => {
		const wrapper = setup({ onLoginClick: jest.fn() });
		expect(wrapper).toMatchSnapshot();
	});

	describe('when the login button is pressed', () => {
		it('calls onLoginClick function', () => {
			const onLoginClick = jest.fn();
			const wrapper = setup({ onLoginClick });
			const header = findByDataTest(wrapper, 'header');
			header.simulate('click');
			expect(onLoginClick).toHaveBeenCalled();
		});
	});
});
