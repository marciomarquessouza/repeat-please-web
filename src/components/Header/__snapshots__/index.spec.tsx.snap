// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Header renders properly 1`] = `
<styled.div>
  <Link
    to="/"
  >
    <Component />
  </Link>
  <styled.div>
    <Component
      data-test="header"
      onClick={[MockFunction]}
      variant="primary"
    >
      login
    </Component>
  </styled.div>
</styled.div>
`;
